.circleci/config.yml

```diff
@@ -215,6 +215,14 @@ jobs:
           # Report but don't fail the build (`|| exit 0`)
           command: npm run typecheck || exit 0
 
+  typecheck-strict:
```

Comment:
We could call it something more descriptive: `typecheck-gutenboarding`?

Reply:
I was considering that, but then decided that we might not make it project-specific but might later add other projects to strictly check here -- vs giving them all a separate CircleCI targets. (My choice is reflected by the [`name`](https://github.com/Automattic/wp-calypso/pull/37945/files#diff-1d37e48f9ceff6d8030570cd36286a61R224), `TypeScript strict typecheck of individual subprojects`)

I don't feel super strong about this -- would you prefer a project-specific target name?

Reply:
+1 for keeping it generic, but a soft +1. Either way is fine, this is pretty easy to change later.

Reply:
Generic seems fine, it's easy enough to change 👍